docs/features/bgp-integration/route-advertisements.md

```diff
@@ -0,0 +1,620 @@
+    advertise: "true"
+spec:
+  namespaceSelector:
+    matchLabels:
+      network: extranet
+  network:
+    topology: Layer3
+    layer3:
+      role: Primary
+      subnets:
+      - cidr: "22.100.0.0/16"
+        hostSubnet: 24
+```
+
+This example is similar to the previous one with the exception that the BGP
+peering session is configured to happen over VRF `extranet`:
+
+```yaml
+apiVersion: frrk8s.metallb.io/v1beta1
+kind: FRRConfiguration
+metadata:
+  labels:
+    use-for-advertisements: extranet
+  name: receive-filtered-extranet
+  namespace: frr-k8s-system
+spec:
+  nodeSelector: {}
+  bgp:
+    routers:
+    - asn: 64512
+      neighbors:
+      - address: 192.168.221.3
+        asn: 64512
+        disableMP: true
+        toReceive:
+          allowed:
+            mode: filtered
+            prefixes:
+            - prefix: 172.20.0.0/16
+      vrf: extranet
+```
+
+This will result in the routes being installed in the extranet VRF owned by the
+CUDN of the same name. If `route-advertisements` feature is enabled,
+OVN-Kubernetes will synchronize the BGP routes installed on a network's owned
+VRF to that OVN pod network's gateway router and hence will be used for the
+egress traffic of the pods on that network.
+
+> [!NOTE]
+> As long as the name of the CUDN is less than 16 characters, the corresponding
+> VRF name for the network will have the same name. Otherwise the name will be
+> pseudo-randomly generated and not easy to predict. Future enhancements will
+> allow for the VRF name to be configurable.
+
+A typical scenario is to import installed BGP routes from the default VRF to a
+CUDN. This can be achieved with:
+
+```yaml
+apiVersion: frrk8s.metallb.io/v1beta1
+kind: FRRConfiguration
+metadata:
+  labels:
+    use-for-advertisements: default
+  name: import-extranet
+  namespace: frr-k8s-system
+spec:
+  nodeSelector: {}
+  bgp:
+    routers:
+    - asn: 64512
+      imports:
+      - vrf: default
+      vrf: extranet
+```
+
+> [!NOTE]
+> If you export routes for a CUDN over the default VRF as detailed on the next
+> sections, installed BGP routes in the default VRF are imported to the CUDN
+> automatically and this configuration is not necessary.
+
+### Export routes to the default pod network
+
+Assuming the `FRRConfiguration` examples that have been used previously, this
+example would advertise routes to the default pod network and its egress IPs:
+
+```yaml
+apiVersion: k8s.ovn.org/v1
+kind: RouteAdvertisements
+metadata:
+  name: default
+spec:
+  targetVRF: default
+  advertisements:
+  - PodNetwork
+  - EgressIP
+  nodeSelector: {}
+  frrConfigurationSelector:
+    matchLabels:
+      use-for-advertisements: default
+  networkSelectors:
+  - networkSelectionType: DefaultNetwork
+```
+
+This would advertise routes for the pod network to the BGP peers as defined on
+the selected `FRRConfiguration` instances; and make the necessary changes to
+correctly handle N/S traffic directly addressing IPs of that network.
```

Comment:
Shall we also describe the behavior when a RA selects multiple `FRRConfiguration`?

Reply:
That's already described in detail in the `FRRConfiguration instances generated by OVN-Kubernetes` section. Anything specifically in mind?